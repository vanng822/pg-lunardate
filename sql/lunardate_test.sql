DROP TABLE IF EXISTS lunartest;
DROP EXTENSION IF EXISTS lunardate;
CREATE EXTENSION lunardate;
CREATE TABLE lunartest(
  id serial PRIMARY KEY,
  from_date lunardate,
  to_date lunardate
);
SET DateStyle=ISO;
insert into lunartest(from_date, to_date) values('2018-10-10', '2019-01-13');
insert into lunartest(from_date, to_date) values('2018-11-11', '2018-12-01');
select * from lunartest;
select id, from_date as lunar_date, from_date::date as solar_date from lunartest;
select * from lunartest where from_date = '2018-11-11';
select '2018-11-11'::lunardate::date;
select '2018-12-17'::date::lunardate;
