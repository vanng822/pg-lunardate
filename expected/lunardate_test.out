DROP TABLE IF EXISTS lunartest;
NOTICE:  table "lunartest" does not exist, skipping
DROP EXTENSION IF EXISTS lunardate;
NOTICE:  extension "lunardate" does not exist, skipping
CREATE EXTENSION lunardate;
CREATE TABLE lunartest(
  id serial PRIMARY KEY,
  from_date lunardate,
  to_date lunardate
);
SET DateStyle=ISO;
insert into lunartest(from_date, to_date) values('2018-10-10', '2019-01-13');
insert into lunartest(from_date, to_date) values('2018-11-11', '2018-12-01');
select * from lunartest;
 id | from_date  |  to_date   
----+------------+------------
  1 | 2018-10-10 | 2019-01-13
  2 | 2018-11-11 | 2018-12-01
(2 rows)

select id, from_date as lunar_date, from_date::date as solar_date from lunartest;
 id | lunar_date | solar_date 
----+------------+------------
  1 | 2018-10-10 | 2018-11-16
  2 | 2018-11-11 | 2018-12-17
(2 rows)

select * from lunartest where from_date = '2018-11-11';
 id | from_date  |  to_date   
----+------------+------------
  2 | 2018-11-11 | 2018-12-01
(1 row)

